apps/docs/src/components/TopBar/TopBar.svelte

```diff
@@ -30,34 +58,89 @@
+    <Menu.Item>
+      <ActionIcon variant="default" on:click={changeModalState} size={30}>
+        <MagnifyingGlass />
+      </ActionIcon>
+    </Menu.Item>
 	</Menu>
 {:else}
-	<div
-		style={`padding-right: ${$mobile ? '0rem' : '0.75rem'}; padding-top: ${
-			$mobile ? '0rem' : '0.75rem'
-		}`}
-	>
-		<ul>
-			{#each config.buttons as { title, props, icon }}
-				<li>
-					<Tooltip withArrow label={title}>
-						<ActionIcon root="a" {...props} radius="md" size="lg">
-							<svelte:component this={icon} size={20} />
-						</ActionIcon>
-					</Tooltip>
-				</li>
-			{/each}
-			<li>
-				<Tooltip withArrow label="Experimental Theme Toggle">
-					<ActionIcon size="lg" color="dark" variant="outline" on:click={toggleTheme} radius="md">
-						{#if $colorScheme === 'light'}
-							<Moon size={20} />
-						{:else}
-							<Sun size={20} />
-						{/if}
-					</ActionIcon>
-				</Tooltip>
-			</li>
-		</ul>
-	</div>
+  <ul style={`padding-right: 0.75rem`} use:hotkey={[['mod+k', () => changeModalState()]]}>
+    <li class="searchBox">
+      <Box css={{ width: "100%", display: "flex", alignItems: "center", justifyContent: "space-between"}} on:click={changeModalState}>
+        <div style="display: flex; align-items: center">
+          <MagnifyingGlass size={25} />
+          <p style="margin-left: 0.5rem; font-size: 1.1rem">Search</p>
+        </div>
+        <div>
+          <Kbd>{navigator.platform === "MacIntel" ? "⌘" : "Ctrl"}</Kbd> + <Kbd>K</Kbd>
+        </div>
+      </Box>
+    </li>
+    {#each config.buttons as { title, props, icon }}
+      <li>
+        <Tooltip withArrow label={title}>
+          <ActionIcon root="a" {...props} radius="md" size="lg">
+            <svelte:component this={icon} size={20} />
+          </ActionIcon>
+        </Tooltip>
+      </li>
+    {/each}
+    <li>
+      <Tooltip withArrow label="Experimental Theme Toggle">
+        <ActionIcon size="lg" color="dark" variant="outline" on:click={toggleTheme} radius="md">
+          {#if $colorScheme === 'light'}
+            <Moon size={20} />
+          {:else}
+            <Sun size={20} />
+          {/if}
+        </ActionIcon>
+      </Tooltip>
+    </li>
+  </ul>
 {/if}
+
+<Modal opened={modalOpened} on:close={changeModalState} title="SvelteUI Docs" overflow="inside">
+	<TextInput
+		placeholder="Search..."
+		bind:value={searchTerm}
+		on:input={onSearchValueInput}
+    autocomplete="off"
+	>
+		<svelte:fragment slot="rightSection">
+			<MagnifyingGlass color="#228be6" size={20} />
+		</svelte:fragment>
+	</TextInput>
+	{#if searchTerm.length === 0}
+    {#if recentSearches.length > 0}
+      <p style="font-size: 0.9rem; margin-top: 0.8rem">Recent searches: </p>
+      {#each recentSearches as recentSearch}
+        <a href={recentSearch.link} style={`text-decoration: none`} on:click={() => addSearch(recentSearch)}>
+          <Paper class="searchTerm" withBorder>
+            <p style="margin: 0; font-size: 1rem">{recentSearch.title}</p>
```

Comment:
I think it would be better to move all these inline styles to the stylesheet of this component and give them classes. What do you think?

Reply:
I agree with this